packages/swapper/src/swappers/zrx/utils/helpers/helpers.ts

```diff
@@ -69,56 +69,76 @@ export const getAllowanceRequired = async ({
   web3,
   erc20AllowanceAbi
 }: GetAllowanceRequiredArgs): Promise<BigNumber> => {
-  if (sellAsset.symbol === 'ETH') {
-    return bn(0)
-  }
+  try {
+    if (sellAsset.assetId === 'eip155:1/slip44:60') {
+      return bn(0)
+    }
 
-  const ownerAddress = receiveAddress
-  const spenderAddress = allowanceContract
-  const tokenId = sellAsset.tokenId
+    const ownerAddress = receiveAddress
+    const spenderAddress = allowanceContract
+    const tokenId = sellAsset.tokenId
 
-  if (!ownerAddress || !spenderAddress || !tokenId) {
-    throw new SwapError(
-      'getAllowanceRequired - receiveAddress, allowanceContract and tokenId are required'
-    )
-  }
+    if (!tokenId) {
+      throw new ZrxSwapError('[getAllowanceRequired] - sellAsset.tokenId is required', {
+        code: SwapErrorTypes.ALLOWANCE_REQUIRED
+      })
+    }
 
-  const allowanceOnChain = await getERC20Allowance({
-    web3,
-    erc20AllowanceAbi,
-    ownerAddress,
-    spenderAddress,
-    tokenId
-  })
-  if (allowanceOnChain === '0') {
-    return bnOrZero(sellAmount)
-  }
-  if (!allowanceOnChain) {
-    throw new SwapError(`No allowance data for ${allowanceContract} to ${receiveAddress}`)
+    const allowanceOnChain = await getERC20Allowance({
+      web3,
+      erc20AllowanceAbi,
+      ownerAddress,
+      spenderAddress,
+      tokenId
+    })
+    if (allowanceOnChain === '0') {
+      return bnOrZero(sellAmount)
+    }
+    if (!allowanceOnChain) {
+      throw new ZrxSwapError(
+        `[getAllowanceRequired] - No allowance data for ${allowanceContract} to ${receiveAddress}`,
+        { code: SwapErrorTypes.ALLOWANCE_REQUIRED }
+      )
+    }
+    const allowanceRequired = bnOrZero(sellAmount).minus(allowanceOnChain)
+    return allowanceRequired.lt(0) ? bn(0) : allowanceRequired
+  } catch (e) {
+    throw new ZrxSwapError('[getAllowanceRequired]', {
+      cause: e,
+      code: SwapErrorTypes.ALLOWANCE_REQUIRED
+    })
   }
-  const allowanceRequired = bnOrZero(sellAmount).minus(allowanceOnChain)
-  return allowanceRequired.lt(0) ? bn(0) : allowanceRequired
 }
 
 export const getUsdRate = async (input: Pick<Asset, 'symbol' | 'tokenId'>): Promise<string> => {
   const { symbol, tokenId } = input
-  if (symbol === 'USDC') return '1' // Will break if comparing against usdc
-  const rateResponse: AxiosResponse<ZrxPriceResponse> = await zrxService.get<ZrxPriceResponse>(
-    '/swap/v1/price',
-    {
-      params: {
-        buyToken: 'USDC',
-        buyAmount: '1000000000', // rate is imprecise for low $ values, hence asking for $1000
-        sellToken: tokenId || symbol
+  try {
+    if (symbol === 'USDC') return '1' // Will break if comparing against usdc
```

Comment:
at least use assetId, we should *only* ever use symbol when talking to zrx

better option - use contract addresses if zrx supports it

best option - if zrx has an endpoint that gives us all the available symbols, add a caip adapter for zrx, e.g.

`type BuyToken = getZrxId(assetId: AssetId) => string`

Reply:
contract address on asset should be what we what we need